Assignment1-Yuxiang/APIBinarySearch/main.cpp

```diff
@@ -0,0 +1,33 @@
+#include <iostream>
+#include <vector>
+
+/// @return: `true` if value is in [begin, end)
+/// @requirements
+///   - iterator can randomly access memory
+///   - the sequence should be sorted in non-descending order
+template<typename Iterator, typename Value>
+bool binarySearch(Iterator begin, Iterator end, Value &&value) {
+    while (begin < end) {
+        auto mid = begin + (end - begin) / 2;
+        if (*mid == value) { return true; }
+        if (*mid > value) { end = mid; }
+        else { begin = mid + 1; }
+    }
+
+    return false;
+}
+
+const std::pair<std::vector<int>, int> testCases[] = {
```

Comment:
I don't want to see any C style arrays. That where we always got segmentation faults or transboundary errors.
Say hello to `std::array`, it's also a zero-overhead container(and safer).